nucypher/blockchain/eth/actors.py

```diff
@@ -878,19 +879,30 @@ def current_stake(self) -> NU:
         """The total number of staked tokens, i.e., tokens locked in the current period."""
         return self.locked_tokens(periods=0)
 
-    def stakes_filtered_by_status(self, parent_status: Stake.Status) -> Iterable[Stake]:
-        """Returns stakes for this staker which have specified or child status."""
+    def filtered_stakes(self,
+                        parent_status: Stake.Status = None,
+                        filter_function: Callable[[Stake], bool] = None
+                        ) -> Iterable[Stake]:
+        """Returns stakes for this staker which filtered by specified or child status or by provided function."""
+        if not parent_status and not filter_function:
+            raise ValueError(f"Pass parent status or filter function or both.")
 
         # Read once from chain and reuse these values
         staker_info = self.staking_agent.get_staker_info(self.checksum_address)  # TODO related to #1514
         current_period = self.staking_agent.get_current_period()                 # TODO #1514 this is online only.
 
-        stakes = (stake for stake in self.stakes if stake.status(staker_info, current_period).is_child(parent_status))
+        stakes = (stake for stake in self.stakes
+                  if (parent_status is None or stake.status(staker_info, current_period).is_child(parent_status)) and
```

Comment:
woah, wordy!

Reply:
yeah, is there way to make it prettier?

Reply:
Somehow, by giving it a name by using a named variable at minimum, or possibly a function/method composed elsewhere?

Reply:
how about extract this to for loop?

Reply:
👍 A for loop would be cleaner

Reply:
✔️